### PR TITLE
Switch to JDK 14 for releasing and CI build

### PR DESCRIPTION
**Pull request by minwoox**
net.sf:proguard-gradle:6.2.2 -> com.guardsquare:proguard-gradle:7.0.0

### CODE REVIEW

**Comment by minwoox** [comment]
Appveyor does not have JDK 15 yet. 😢 
https://www.appveyor.com/docs/windows-images-software/#java

File: bucket4j/src/main/java/com/linecorp/armeria/server/throttling/bucket4j/TokenBucket.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public static TokenBucket of(long limit, Duration period) {
      * Defines throttling configuration comprised of zero or more bandwidth limits in accordance to
      * token-bucket algorithm.
      *
-     * <h3>Multiple bandwidths:</h3>
+     * <h2>Multiple bandwidths:</h2>
```
Comment (adriancole) [inline bucket4j/src/main/java/com/linecorp/armeria/server/throttling/bucket4j/TokenBucket.java:81]: is JDK 15 strict about h3 without h2?

Comment (minwoox) [reply bucket4j/src/main/java/com/linecorp/armeria/server/throttling/bucket4j/TokenBucket.java:81]: Yes, the build fails if headings are used without the order.

File: core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java [hunk 2]
```diff
@@ -433,7 +433,7 @@ public ClientFactoryBuilder idleTimeoutMillis(long idleTimeoutMillis) {
      * Sets the PING interval in milliseconds.
      * When neither read nor write was performed for the given {@code pingIntervalMillis},
      * a <a href="https://httpwg.org/specs/rfc7540.html#PING">PING</a> frame is sent for HTTP/2 or
-     * an <a herf="https://tools.ietf.org/html/rfc7231#section-4.3.7">OPTIONS</a> request with an asterisk ("*")
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java:436]: Herf time!

**Review by trustin — APPROVED** [review]
LGTM. I think it's fine to leave Windows builds in Java 14 until AppVeyor updates the image.

**Comment by trustin** [comment]
Ohh, I see some build failures related with Javadoc, though. Gotta fix that before moving forward.

**Comment by minwoox** [comment]
```
Execution failed for task ':core:trimShadedJar'.
> java.io.IOException: Can't read [/.../java/15.0.0.hs-adpt/jmods/java.security.sasl.jmod] (Can't process class [module-info.class] (Unsupported version number [59.0] (maximum 58.0, Java 14)))
```
I have to dig in more but it seems like proguard does not work well with JDK 15.

**Comment by minwoox** [comment]
Let me just use JDK 14 for now because Appveyor also does not support JDK 15.

File: spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/spring/DocServiceConfigurator.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
 import com.linecorp.armeria.server.docs.DocServiceBuilder;
 
 /**
- * Configures a {@link DocService} built by {@link ArmeriaAutoConfiguration}.
+ * Configures a {@link DocService} built by Armeria auto configuration.
```
Comment (minwoox) [inline spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/spring/DocServiceConfigurator.java]: This class is copied to `boot2-webflux-autoconfigure` and it cannot connect to the class there.

File: core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java (hunk 2 above)
Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java:436]: 😱

**Review by ikhoon — APPROVED** [review]
Thanks! @minwoox

File: build.gradle [hunk 4]
```diff
@@ -207,9 +207,9 @@ allprojects {
     }
 }
 
-// Require to use JDK 13 when releasing.
+// Require to use JDK 15 when releasing.
```
Comment (KarboniteKream) [inline build.gradle:210]: This is still 15 😁

Comment (minwoox) [reply build.gradle:210]: Oopsy. 🤣

**Pull request merged**